PythonRegression/tests/features/steps/api_test_steps.py

```diff
@@ -124,23 +124,93 @@ def compare_thread_return(step,apiCall):
     logger.info('Responses match')
 
 
-@step(r'GTTA is called (\d+) times on "([^"]+)"')
-def spam_call_gtta(step,numTests,node):
-    apiCall = 'getTransactionsToApprove' 
-    world.config['apiCall'] = apiCall
+#TODO: Break these functions down into utilities
+@step(r'GTTA is called (\d+) times')
+def spam_call_gtta(step,numTests):
+    start = time()
+    node = next(iter(world.machine['nodes']))
+    apiCall = 'getTransactionsToApprove'
+
     world.config['nodeId'] = node
-    
-    api = api_utils.prepare_api_call(node)
-    logging.info('Calls being made to %s',node)
-    responseVal = []
-    for i in range(int(numTests)):
-        logging.debug("Call %d made", i+1)
+    world.config['apiCall'] = apiCall
+
+    nodes = {}
+    queue = {}
+    threads = {}
+    thread_status = {}
+
+    for current_node in world.machine['nodes']:
+        api = api_utils.prepare_api_call(current_node)
+        nodes[current_node] = api
+        queue[current_node] = "Not Running"
+
+
+    def run_call(node,api):
+        logger.debug('Running Thread on {}'.format(node))
+        queue[node] = "Running"
         response = api.get_transactions_to_approve(depth=3)
+        logger.debug('Response complete, placing thread back in queue')
         responseVal.append(response)
-        
+        queue[node] = "Not Running"
+
+    logging.info('Calls being made to %s',node)
+    responseVal = []
+
+    i = 0
+    logger.info(int(numTests))
+    while i < int(numTests):
+        logger.info('Status: {} / {}'.format(i,int(numTests)))
+        for thread in queue:
+            if queue[thread] == "Not Running" and i < int(numTests):
+                new_thread = threading.Thread(name=thread,target=run_call,args=(thread,nodes[thread]))
+                new_thread.start()
+                threads[thread] = new_thread
+                i += 1
+        sleep(1)
+
+        logger.debug('Joining threads')
+        for thread_id in threads:
+            thread = threads[thread_id]
+            thread.join(1)
+            alive = thread.is_alive()
+            thread_status[thread_id] = alive
+
+        logger.debug('Pruning threads')
+        for thread in thread_status:
+            if thread_status[thread] == False and thread in threads:
+                threads.pop(thread)
+
+
+    run = True
+    while run == True:
+        logger.info(len(threads))
+        thread_count = 0
+        for thread in threads:
+            current_thread = threads[thread]
+            alive = current_thread.is_alive()
+            sleep(1)
+            if alive:
+                logger.info('Waiting')
+                sleep(3)
```

Comment:
Using `sleep` is usually a bad sign...
Remember, if you ran all threads before it is ok to block the main thread! This is because the background requests will continue to run.